src/utils/model_utils.py

```diff
@@ -70,29 +74,91 @@ def train(self,
      float]:
         """TODO: generate docstring
         """
+        print("started training with dset ", self.dset)
         model.train()
-        train_loss = 0
-        correct = 0
-        for batch_idx, (data, target) in enumerate(dloader):
-            data, target = data.to(device), target.to(device)
-            optim.zero_grad()
-            position = kwargs.get("position", 0)
-            output = model(data, position=position)
-            if kwargs.get("apply_softmax", False):
-                output = nn.functional.log_softmax(
-                    output, dim=1)  # type: ignore
-            loss = loss_fn(output, target)
-            loss.backward()
-            optim.step()
-            train_loss += loss.item()
-            pred = output.argmax(dim=1, keepdim=True)
-            # view_as() is used to make sure the shape of pred and target are
-            # the same
-            if len(target.size()) > 1:
-                target = target.argmax(dim=1, keepdim=True)
-            correct += pred.eq(target.view_as(pred)).sum().item()
+
```

Comment:
Can you please make this function modular? It is going to get worse if we add a few more tasks with totally different evaluation. I recommend creating two functions - one for pascal and one for classification (that should be called when else is executed)

Reply:
:0 done